src/components/node-list/node-list-items.js

```diff
@@ -188,7 +244,7 @@ const compareEnabledThenAlpha = (itemA, itemB) => {
  */
 const compareEnabledThenType = (itemA, itemB) => {
   const byEnabledTag = Number(itemA.disabled_tag) - Number(itemB.disabled_tag);
-  const nodeTypeIDs = ['task', 'data', 'parameters'];
+  const nodeTypeIDs = Object.keys(sidebarElementTypes);
```

Comment:
there isn't any need to declare or reference nodeTypeIDs `sidebarElementTypes` from `config.js` here as `nodeTypes` are being passed into the state of the selectors via the `nodeTypes` value in `node-list/index.js` - we can just simply pass that on as a variable from the selector that references it.

Reply:
I thought the ordering used here is fixed and depends on the sidebar design though? If that's the case then the config seems like a good place to declare that intention?

Reply:
aargh I see your confusion - currently the order of the elements in the tree follows the order of the array declared under `nodes.type` in `normalize-data.js` ( i.e the value of `nodeTypes` in the state passed onto the reducers), and I had set up the order here as an array more so for clarity purposes to reiterate that order ( which in retrospective it might be better if I set the function to take in a `nodeTypeIDs` array instead and pass in the `nodeTypes`)

In looking at the elements declared in config, I do see that it is more so for the purpose of structuring the groups for the sidebar filter, which does make sense to declare it separately instead of reusing `nodeTypes` from state, as the grouping on the filter panel depends on the design. 

In terms of the `nodeTypeIDs` value used here, given that the function to sort by nodetype  is mainly utilised to sort elements for the tree, I would follow the setup by the other tree selectors in using `nodeTypes` from the state as a reference for the order, in case we do add any new node types that will be declared in `normalize-data.js` in the future. ( It might be even better to bring this forward that, instead of using a hardcoded object in config to declare the element types, we can also instead reference the node types declared in the state to set up the `sidebarElementTypes`. )

I understand this might be a bit of refactor work, I am also happy to do it in the focus mode PR if you wish to avoid blocking this PR :smiley

Reply:
Ah ok then happy to leave it to you, but I think probably worth adding a comment explaining the order and maybe pulling it from a named constant somewhere, especially if it requires being in sync with something else.